### PR TITLE
drivers: adc: shell: fix - read result is signed

### PR DESCRIPTION
**Pull request by nixward**
Previously negative read results were being printed by the shell as large uint16_t.

Also adds `differential` command.

### CODE REVIEW

File: drivers/adc/adc_shell.c [hunk 1]
```diff
@@ -337,6 +337,12 @@ static int cmd_adc_print(const struct shell *sh, size_t argc, char **argv)
 			   adc->channel_config.acquisition_time,
 			   adc->channel_config.channel_id,
 			   adc->resolution);
+#if CONFIG_ADC_CONFIGURABLE_INPUTS
+	shell_print(sh, "Input positive: %u\n"
+			"Input negative: %u",
+			adc->channel_config.input_positive,
+			adc->channel_config.input_negative);
```
Comment (anangl) [inline drivers/adc/adc_shell.c:344]: Printing of the negative input only makes sense for differential channels.

Comment (nixward) [reply drivers/adc/adc_shell.c]: @anangl thanks, fixed, I've added the differential command.

**Review by anangl — APPROVED** [review]

**Comment by nixward** [comment]
@nandojve would you mind having a look?

**Review by nandojve — APPROVED** [review]
> @nandojve would you mind having a look?

LGTM! I added @attie-argentum to have a look too.

**Comment by attie-argentum** [comment]
Thanks @nandojve - LGTM too.

**Pull request merged**